mutation_reader.cc

```diff
@@ -1696,14 +1696,13 @@ reader_lifecycle_policy::try_resume(reader_concurrency_semaphore::inactive_read_
 
 flat_mutation_reader make_multishard_combining_reader(
         shared_ptr<reader_lifecycle_policy> lifecycle_policy,
-        const dht::i_partitioner& partitioner,
         schema_ptr schema,
         const dht::partition_range& pr,
         const query::partition_slice& ps,
         const io_priority_class& pc,
         tracing::trace_state_ptr trace_state,
         mutation_reader::forwarding fwd_mr) {
-    return make_flat_mutation_reader<multishard_combining_reader>(std::move(lifecycle_policy), partitioner, std::move(schema), pr, ps, pc,
+    return make_flat_mutation_reader<multishard_combining_reader>(std::move(lifecycle_policy), schema->get_partitioner(), std::move(schema), pr, ps, pc,
```

Comment:
I think we should go all the way and remove `multishard_combining_reader::_partitioner` too.

Reply:
done.